src/test/java/com/exasol/spark/BaseIntegrationSetup.java

```diff
@@ -0,0 +1,110 @@
+import com.exasol.containers.ExasolDockerImageReference;
+import com.exasol.dbbuilder.dialects.exasol.ExasolObjectFactory;
+import com.exasol.dbbuilder.dialects.exasol.ExasolSchema;
+
+/**
+ * A base integration test class with Exasol docker container setup.
+ */
+public class BaseIntegrationSetup {
+    private static final Logger LOGGER = Logger.getLogger(BaseIntegrationSetup.class.getName());
+    private static final String DEFAULT_DOCKER_IMAGE = "7.1.19";
+
+    @Container
+    private static final ExasolContainer<? extends ExasolContainer<?>> EXASOL = new ExasolContainer<>(
+            getExasolDockerImage()).withReuse(true);
+
+    protected static Connection connection;
+    protected static ExasolObjectFactory factory;
+    protected static ExasolSchema exasolDatabase;
+    protected static SparkSession spark;
+
+    @BeforeAll
+    public static void beforeAll() throws SQLException {
+        EXASOL.purgeDatabase();
+        connection = EXASOL.createConnection();
+        factory = new ExasolObjectFactory(connection);
+        spark = SparkSessionProvider.getSparkSession(createSparkConfiguration());
+        createDatabase("DEFAULT_SCHEMA");
+    }
+
+    @AfterAll
+    public static void afterAll() throws SQLException {
+        dropDatabase();
+        connection.close();
+        spark.close();
+    }
+
+    private static void createDatabase(final String exasolSchemaName) {
+        LOGGER.fine(() -> "Creating a new schema '" + exasolSchemaName + '"');
+        dropDatabase();
+        exasolDatabase = factory.createSchema(exasolSchemaName);
+    }
+
+    private static void dropDatabase() {
+        if (exasolDatabase != null) {
+            LOGGER.fine(() -> "Dropping schema '" + exasolDatabase.getName() + '"');
+            exasolDatabase.drop();
+            exasolDatabase = null;
+        }
+    }
+
+    public Map<String, String> getOptionsMap() {
+        final Map<String, String> map = Stream.of(new String[][] { //
+                { "host", EXASOL.getDockerNetworkInternalIpAddress() }, //
+                { "port", EXASOL.getDefaultInternalDatabasePort() + "" }, //
+                { "username", EXASOL.getUsername() }, //
+                { "password", EXASOL.getPassword() }, //
+                { "jdbc_url", EXASOL.getJdbcUrl() }, //
+        }).collect(Collectors.toMap(e -> e[0], e -> e[1]));
+        if (imageSupportsFingerprint()) {
+            map.put("fingerprint", getFingerprint());
+        } else {
+            map.put("jdbc_options", "validateservercertificate=0");
+        }
+        LOGGER.fine(() -> "Prepared options '" + map.toString() + "'.");
+        return map;
+    }
+
+    private String getFingerprint() {
+        return EXASOL.getTlsCertificateFingerprint().get();
+    }
+
+    private boolean imageSupportsFingerprint() {
+        final ExasolDockerImageReference image = EXASOL.getDockerImageReference();
+        return (image.getMajor() >= 7) && (image.getMinor() >= 1);
```

Comment:
This won't work for Exasol 8. But do we still need to support 7.0, could we remove the check completely?

Reply:
Yes, good idea, hope 7.0 is not used but we never know.